src/client/components/search-component/index.jsx

```diff
@@ -0,0 +1,213 @@
+                </div>
+            </Form> :
+            <Form className='search-panel'>
+            <Header as='h2'>Search</Header>
+            <Form.Field className='destination'>
+                <label>Destination/property name:</label>
+                <input
+                    name='destination'
+                    placeholder='Where are you going?'
+                    onChange={this.handleChange.bind(this)}
+                />
+            </Form.Field>
+            <Form.Field className='check-in-out'>
+                <label>Check-in and Check-out date</label>
+                <DatesRangeInput
+                    closable
+                    minDate={moment()}
+                    dateFormat='ddd D MMM'
+                    popupPosition='bottom center'
+                    icon='calendar alternate outline'
+                    iconPosition='left'
+                    placeholder='Check-in - Check-out'
+                    name='date'
+                    value={this.state.date}
+                    onChange={this.handleChange.bind(this)}
+                />
+            </Form.Field>
+            <div className='room-options'>
+                <div
+                    id='room-selector'
+                    className='room-selector'
+                >
+                    <Form.Field>
+                        <Dropdown
+                            fluid
+                            selection
+                            name='adults'
+                            text={`${this.state.adults} ${this.state.adults === 1 ? 'Adult' : 'Adults'}`}
+                            options={selectOptions}
+                            value={this.state.adults}
+                            onChange={this.handleChange.bind(this)}
+                        />
+                    </Form.Field>
+                    <Form.Group inline>
+                        <Dropdown
+                            selection
+                            name='children'
+                            text={`${this.state.children === 0 ? 'No Children' : `${this.state.children} ${this.state.children === 1 ? 'Child' : 'Children'}`}`}
+                            options={childrenOptions}
+                            value={this.state.children}
+                            onChange={this.handleChange.bind(this)}
+                        />
+                        <Dropdown
+                            selection
+                            name='rooms'
+                            text={`${this.state.rooms} ${this.state.rooms === 1 ? 'Room' : 'Rooms'}`}
+                            options={selectOptions}
+                            value={this.state.rooms}
+                            onChange={this.handleChange.bind(this)}
+                        />
+                    </Form.Group>
+                </div>
+            </div>
+            <div className='btn-wrp'>
+                <Button type='submit' content='Search' primary/>
```

Comment:
Even though your component isn't fully interactive yet, you can already add a function-prop, something like `onClick={this.props.onSubmit}`